generator/region_meta.cpp

```diff
@@ -0,0 +1,115 @@
+#include "region_meta.hpp"
+
+#include "coding/reader.hpp"
+
+#include "platform/platform.hpp"
+
+#include "3party/jansson/myjansson.hpp"
+
+namespace
+{
+int8_t ParseHolidayReference(string const & ref)
+{
+  if (ref == "easter")
+    return feature::RegionData::PHReference::PH_EASTER;
+  if (ref == "orthodox easter")
+    return feature::RegionData::PHReference::PH_ORTHODOX_EASTER;
+  if (ref == "victoriaDay")
+    return feature::RegionData::PHReference::PH_VICTORIA_DAY;
+  if (ref == "canadaDay")
+    return feature::RegionData::PHReference::PH_CANADA_DAY;
+  return 0;
+}
+}  // namespace
+
+namespace feature
+{
+bool ReadRegionDataImpl(string const & countryName, RegionData & data)
+{
+  try
+  {
+    auto reader = GetPlatform().GetReader(COUNTRIES_META_FILE);
+    string buffer;
+    reader->ReadAsString(buffer);
+    my::Json root(buffer.data());
+
+    json_t * jsonData = nullptr;
+    my::FromJSONObjectOptionalField(root.get(), countryName, jsonData);
+    if (!jsonData)
+      return false;
+
+    vector<string> languages;
+    my::FromJSONObjectOptionalField(jsonData, "languages", languages);
+    if (!languages.empty())
+      data.SetLanguages(languages);
+
+    string driving;
+    my::FromJSONObjectOptionalField(jsonData, "driving", driving);
+    if (driving == "l" || driving == "r")
+      data.Set(RegionData::Type::RD_DRIVING, driving);
+
+    string timezone;
+    my::FromJSONObjectOptionalField(jsonData, "timezone", timezone);
+    if (!timezone.empty())
+      data.Set(RegionData::Type::RD_TIMEZONE, timezone);
+
+    string housenames;
+    my::FromJSONObjectOptionalField(jsonData, "housenames", housenames);
+    if (housenames == "y")
+      data.Set(RegionData::Type::RD_ALLOW_HOUSENAMES, housenames);
+
+    // Public holidays: an array of arrays of [string/number, number].
+    // See https://github.com/opening-hours/opening_hours.js/blob/master/docs/holidays.md
+    vector<json_t *> holidays;
+    my::FromJSONObjectOptionalField(jsonData, "holidays", holidays);
+    for (json_t * holiday : holidays)
+    {
+      if (!json_is_array(holiday) || json_array_size(holiday) != 2)
+        MYTHROW(my::Json::Exception, ("Holiday must be an array of two elements in", countryName));
+      json_t * reference = json_array_get(holiday, 0);
+      int8_t refId = 0;
+      if (json_is_integer(reference))
+        refId = json_integer_value(reference);
+      else if (json_is_string(reference))
+        refId = ParseHolidayReference(string(json_string_value(reference)));
+      else
+        MYTHROW(my::Json::Exception,
+                ("Holiday month reference should be either a string or a number in", countryName));
+      if (refId <= 0)
+        MYTHROW(my::Json::Exception, ("Incorrect month reference in", countryName));
```

Comment:
No: it can be up to 90 theoretically. There are special values for easter, victoria day etc, and also for first monday and last weekdays of each month.